scripts/migrate_assets.js

```diff
@@ -318,39 +321,50 @@ var migrateLink = function(link, callback) {
  * @param  {Function}         callback            Standard callback function
  */
 var migrateFile = function(file, callback) {
-  // Download the file to a temporary folder
-  request(file.download_url).on('response', function(res) {
-    // Extract the name of the file
-    var disposition = contentDisposition.parse(res.headers['content-disposition']);
-    var filename = disposition.parameters.filename;
-    var path = os.tmpdir() + filename;
-
-    // Function that will clean up the temporary file
-    var cleanTempFile = function(err) {
-      fs.unlink(path, function() {
-        return callback(err);
-      });
-    };
+  var downloadDir = path.join(os.tmpdir(), processStartTime.toString());
+  fs.stat(downloadDir, function(err, stat) {
+    // Download the file to a temporary folder
+    if (err !== null) {
+      if (err.code === 'ENOENT') {
+        fs.mkdirSync(downloadDir);
+      } else {
+        log.error({downloadDir: downloadDir, errorCode: err.code}, 'Failed to create temp directory during file migration');
+        callback(err);
```

Comment:
@johncrossman Does this need to be `return callback(err)` so that code farther down doesn't execute?